services/collaboration/pkg/internal/app/wopiinfo.go

```diff
@@ -0,0 +1,137 @@
+	// Logs for this endpoint will be covered by the access log. We can't extract
+	// more info
+	http.Error(w, http.StatusText(http.StatusTeapot), http.StatusTeapot)
+}
+
+// CheckFileInfo returns information about the requested file and capabilities of the wopi server
+// https://docs.microsoft.com/en-us/microsoft-365/cloud-storage-partner-program/rest/files/checkfileinfo
+func CheckFileInfo(app *DemoApp, w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	wopiContext, _ := WopiContextFromCtx(ctx)
+
+	statRes, err := app.gwc.Stat(ctx, &providerv1beta1.StatRequest{
+		Ref: &wopiContext.FileReference,
+	})
+	if err != nil {
+		app.Logger.Error().
+			Err(err).
+			Str("FileReference", wopiContext.FileReference.String()).
+			Str("ViewMode", wopiContext.ViewMode.String()).
+			Str("Requester", wopiContext.User.GetId().String()).
+			Msg("CheckFileInfo: stat failed")
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	if statRes.Status.Code != rpcv1beta1.Code_CODE_OK {
+		app.Logger.Error().
+			Str("FileReference", wopiContext.FileReference.String()).
+			Str("ViewMode", wopiContext.ViewMode.String()).
+			Str("Requester", wopiContext.User.GetId().String()).
+			Str("StatusCode", statRes.Status.Code.String()).
+			Str("StatusMsg", statRes.Status.Message).
+			Msg("CheckFileInfo: stat failed with unexpected status")
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	fileInfo := FileInfo{
+		OwnerID:           statRes.Info.Owner.OpaqueId + "@" + statRes.Info.Owner.Idp,
+		Size:              int64(statRes.Info.Size),
+		Version:           statRes.Info.Mtime.String(),
+		BaseFileName:      path.Base(statRes.Info.Path),
+		BreadcrumbDocName: path.Base(statRes.Info.Path),
+		// to get the folder we actually need to do a GetPath() request
+		//BreadcrumbFolderName: path.Dir(statRes.Info.Path),
+
+		UserCanNotWriteRelative: true,
+
+		HostViewUrl: wopiContext.ViewAppUrl,
+		HostEditUrl: wopiContext.EditAppUrl,
+
+		//EnableOwnerTermination: true,  // enable only for collabora? wopivalidator is complaining
+		EnableOwnerTermination: false,
+
+		SupportsExtendedLockLength: true,
+
+		SupportsGetLock: true,
+		SupportsLocks:   true,
+	}
+
+	switch wopiContext.ViewMode {
+	case appproviderv1beta1.ViewMode_VIEW_MODE_READ_WRITE:
+		fileInfo.SupportsUpdate = true
+		fileInfo.UserCanWrite = true
+
+	case appproviderv1beta1.ViewMode_VIEW_MODE_READ_ONLY:
+		// nothing special to do here for now
+
+	case appproviderv1beta1.ViewMode_VIEW_MODE_VIEW_ONLY:
+		fileInfo.DisableExport = true
+		fileInfo.DisableCopy = true
+		fileInfo.DisablePrint = true
+	}
+
+	// user logic from reva wopi driver #TODO: refactor
+	var isPublicShare bool = false
+	if wopiContext.User != nil {
+		if wopiContext.User.Id.Type == userv1beta1.UserType_USER_TYPE_LIGHTWEIGHT {
+			fileInfo.UserID = statRes.Info.Owner.OpaqueId + "@" + statRes.Info.Owner.Idp
```

Comment:
only a limited set of characters is allowed - maybe base64 encode it ....

see https://learn.microsoft.com/en-us/microsoft-365/cloud-storage-partner-program/rest/files/checkfileinfo/checkfileinfo-response#requirements-for-user-identity-properties

Reply:
I've used hex encoding. Both base32 and base64 use padding chars (`=` seems to be standard, not sure if it could be different), and even though we could remove the padding, it seems a bad idea.
Hex encoding should work without problems and there should be enough builtin support everywhere to decode the string if needed.